### DISCUSSION TITLE
Custom Resolver missing nested object values

### DISCUSSION THREAD

**Discussion by HellFirePvP** [nuwave/lighthouse#2586] [Q&A]
Hey, i'm uncertain how to properly phrase the question/title.
So i have a larger Datastructure and i have a entrypoint into it via a custom resolver such as `listEntries(name: xyz, year: 2000)` that then returns entries. That works. However, when i try to extend my result via a query like this:
```
listEntries(name: xyz) {
    id,
    name,
    parent {
        name,
        value
    }
}
```
However, it seems that it doesn't resolve the 'parent' at all. Any value inside it is null, suggesting the object is not joined to the result at all. In my resolver i just create a query with Eloquent's syntax `Entry::query()->select('entry.*')->join(parent)->where parent year = given year` and then return the result via query->get() from the resolver. Am i missing something? Do i somehow have to manually join all tables i want the resolver result to ever potentially be returned from the graphql query?

I'm also more or less forced to use custom resolvers for all resolvers and such, because the underlying permissions logic will work on the individual models if they should be visible to the logged-in user or not, in addition to permissions based on the queries themselves.

Thanks in advance for any suggestions or thoughts.

**Comment by spawnia** [comment]
Does joining `parent` make such a property available on your `Entry` instance, does `$entry->parent` work in PHP code? That is essentially what Lighthouse does when you do not specify another resolver for a field.

**Reply by HellFirePvP** [reply]
Does this work through the defined Associations?
The Entry-model has a `parent()` function with `return $this->belongsTo(Parent::class)`, the actual column in the database being called `parent_id`.

Pardon if i'm not fully asking the right questions either as i'm somewhat new to Eloquent & Laravel itself.

**Reply by spawnia** [reply]
I would recommend to not use joins and to leverage relations instead.
